### ISSUE TITLE
Add Feedback Mechanism

### ISSUE THREAD

**Issue by kknee** [ioos/ioos-us#6]

**Comment by lukecampbell** [comment]
Just a heads up, this was 'supposed' to be a static site. It's not well equipped for any dynamic behavior.

**Comment by kknee** [comment]
This is just adding the ability to send an email from the contact us page.  Send to the same group linked to the feedback form here: http://catalog.ioos.us/help/feedback

**Comment by dpsnowden** [comment]
I disagree with the static site comment. I thought we've been hoping to automate the registration of new data sets by creating a form that initiated the harvesting. Glider DAC has a similar page for data providers. Maybe I don't understand what you mean by static. The ability to collect feedback on every page is very important. ioos.catalog@noaa.gov probably can't cover the feedback for the entire ioos.us.

**Comment by lukecampbell** [comment]
The main application that runs http://ioos.us/ was a small nodeJS project. We made a very important design decision that it would link to dynamic pages and not be dynamic in and of itself. To change this would require us to either add a lot more code to this nodeJS project and make it a dynamic project, or to scrap it and start a web application in python land.  

Right now, it's not a web application, it's just a container that hosts static HTML files.

**Comment by dpsnowden** [comment]
Are you saying that a form collecting feedback is impossible to create under the current design?  I'm still not sure what you mean by dynamic/static. Is an html form populating a database dynamic?

**Comment by kknee** [comment]
@dpsnowden is your concern about a regular feedback mechanism or the ability to register datasets through the site?  This issue was started to deal with the former.

**Comment by lukecampbell** [comment]
So a static HTML file can not, by itself populate, save or do anything with the data. The browser that's reading the HTML can send the data to a server through HTTP commands like POST, or PUT. A server needs to listen for those, accept the data and do something with them. We don't have that last bit, we're just serving content. We can break that, but if we do, in order to choose the right tools, it would be beneficial to us to know the extents of capabilities we're looking to support.

**Comment by kknee** [comment]
@bnfish - this one is still on your plate, please add the ability to provide feedback from the contact us page.  For now the feedback can just be emailed to ioos.catalog@noaa.gov.  We will get fancier later.

**Comment by dpsnowden** [comment]
@kknee my initial purpose for creating this ticket was not focused on the "register data sets" use case.  But, that's another thing we'll need to grapple with in the not too distant future.  Like @lukecampbell said, we should collect several use cases before we do a significant dev interation or even make design choices.  @mwengren is starting catalog planning now and as we discussed, catalog + ioos.us are intertwined. So, here are a couple thoughts for next steps.

- [ ] Current "Contact Us" button on all ioos.us pages is a mailto link to noaa.ioos.webmaster@noaa.gov.  Change to ioos.catalog@noaa.gov for now.  
- [ ] Collect use cases for "dynamic" site features including *all* *.ioos.us subdomains
- [ ] Consider some sort of feedback button like the maroon button on the left side of the [sensor map](http://sensors.ioos.us/#map?lg=02078477-4a39-428c-af4d-ec03e3912636%2C26bbaa25-cea1-4526-b708-88f3d223e072&p=proj3857&b=hybrid)
- [ ] release plan

Thoughts?  

Bottom line is I think ioos.us is going to have to evolve into something more dynamic.  @lukecampbell, my apologies that we didn't catch that design choice early on because these features were always on the long term roadmap in my mind.

**Comment by dpsnowden** [comment]
What is the approach for implementing the feedback mechanism?  I'm starting to receive email from ioos-monitor @ ... but I don't know what generates that email.  Is there a button somewhere?  We've talked around many aspects of this issue but I'm still a little confused about the overall approach for all ioos.us related sites to gather feedback from users.

**Comment by kknee** [comment]
@bnfish as discussed, please add a floating feedback button to the left-side of all ioos.us pages.  For consistency, please make it look like the button here: https://sensors.ioos.us/ and make sure that the pop-up collects the same info.

**Comment by kknee** [comment]
@bnfish how can I review the updates you made?

**Comment by kknee** [comment]
![image](https://user-images.githubusercontent.com/196868/27464132-b362469c-5799-11e7-9afc-c4e5c021bfaf.png)

**Comment by mwengren** [comment]
I think this one has been superseded by #169 , closing.
